tests/codebuild/common.sh

```diff
@@ -52,6 +52,31 @@ function wait_for_crd_status()
   fi
 }
 
+# This function waits for endpoint to reach a specified state
+# varifies the state through aws cli as well
+function hostingdeploymet_wait_for_status(){
+  local crd_namespace="$1"
+  local crd_instance="$2"
+  local enpoint_name="$3"
+  local endpoint_region="$4"
+  local timeout="$5"
+  local desired_status="$6"
+
+  echo "Waiting for endpoint $enpoint_name to be in status $desired_status"
+  wait_for_crd_status "${crd_namespace}" HostingDeployment "${crd_instance}" "${timeout}" "${desired_status}"
+
+  timeout "${timeout}" bash -c \
+    'until [ "$(aws sagemaker describe-endpoint --endpoint-name "$0" --region "$1" --query EndpointStatus --output text)" == "$2" ]; do \
```

Comment:
is this the right apporach? We should be waiting for the k8s resource itself?

What is the root cause of this problem? is the behaviour same if one uses AWS SDK directly instead of operator?

Reply:
there are multiple things which need to be in sync for this test 

this eliminates the possibility of having out of sync status from crd and cli (just for good measure)